axon/src/main/java/com/baeldung/axon/gui/OrderRestEndpoint.java

```diff
@@ -91,4 +97,20 @@ public CompletableFuture<Void> shipOrder(@PathVariable("order-id") String orderI
     public CompletableFuture<List<Order>> findAllOrders() {
         return queryGateway.query(new FindAllOrderedProductsQuery(), ResponseTypes.multipleInstancesOf(Order.class));
     }
+
+    @GetMapping("/total-shipped/{product-id}")
+    public Integer totalShipped(@PathVariable("product-id") String productId) {
+        return queryGateway.scatterGather(new TotalProductsShippedQuery(productId), ResponseTypes.instanceOf(Integer.class), 10L, TimeUnit.SECONDS)
+                .reduce(0, Integer::sum);
+    }
+
+    @GetMapping(path = "/order-updates/{order-id}", produces = MediaType.TEXT_EVENT_STREAM_VALUE)
+    public Flux<Order> orderUpdates(@PathVariable("order-id") String orderId) {
+        return subscriptionQuery(new OrderUpdatesQuery(orderId), ResponseTypes.instanceOf(Order.class));
+    }
+
+    private  <Q, R> Flux<R> subscriptionQuery(Q query, ResponseType<R> resultType) {
+        var result = queryGateway.subscriptionQuery(query, resultType, resultType);
+        return result.initialResult().concatWith(result.updates()).doFinally(signal -> result.close());
```

Comment:
Perhaps an explanation in the code in the form of documentation would be nice to the readers? Or will your article contain something about this?

Reply:
I plan to add something, alternatively I could add the reactive extension, but tried to keep additional dependencies to a minimum. I do plan mention the reactive extension.

Reply:
I agree that adding the Reactor Extension might be a bit to much. 
Mentioning shouldn't hurt if it fits the flow of the article I think.